packages/adapter/src/tonclient4/converters/convertGetBlock.ts

```diff
@@ -0,0 +1,60 @@
+import { Address } from '@ton/core';
+import { GetBlockTransactionsResponse, Block, ShardsResponse, RpcInputs, SeqnoSet } from '../types';
+import { convertHexShardToSignedNumberStr } from '../utils'
+
+export function convertGetBlockTransactionsInputs(seqno: number, data: ShardsResponse): RpcInputs<SeqnoSet> {
+  if (!data.ok) {
+    return []
+  }
+
+  return [
+    {
+      method: "getBlockTransactions",
+      params: {
+        workchain: -1,
+        shard: "8000000000000000",
+        seqno,
+      } 
+    },
+    ...data.result.shards.map(it => {
```

Comment:
`
  const shardInputs = data.result.shards.map(({ workchain, shard, seqno }) => ({
    method: "getBlockTransactions",
    params: { workchain, shard, seqno }
  }));
`  
  is more clean (optional)

Reply:
Sure